game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -669,9 +671,8 @@ public void focusGained(final FocusEvent e) {
     final MovePanel movePanel = new MovePanel(data, mapPanel, this);
     actionButtons = new ActionButtons(data, mapPanel, movePanel, this);
 
-    SwingUtilities.invokeLater(() -> mapPanel.addKeyListener(getFullScreenListener()));
+    addKeyBindings(movePanel, this);
     SwingUtilities.invokeLater(() -> mapPanel.addKeyListener(getArrowKeyListener()));
-    SwingUtilities.invokeLater(() -> mapPanel.addKeyListener(movePanel.getCustomKeyListeners()));
     SwingUtilities.invokeLater(() -> mapPanel.addKeyListener(getFlagToggleKeyListener(this)));
```

Comment:
I believe we could simply a lot of code if similar places were migrated as well.
The whole "input map"/key listener thing is super inconsistent over the codebase

Reply:
The other methods are pretty complex, we don't gain quite that much as the logic code in the listeners is a good bit greater than the infrastructure code of the key listener. I decided to leave it alone for now as that migration was a large effort. I had started down that path and it looked like a pretty expensive rabbit hole.

Reply:
I must admit that `getFlagToggleKeyListener` is a mess.
This was one of my first contributions to the TripleA project, and it basically keeps a Thread alive as long as the key is held down to make sure that we get some sort of "key up" event even if the window loses focus.

Probably not the best idea to realize that but it worked I guess 🤷‍♂